cylc/flow/simulation.py

```diff
@@ -166,46 +196,59 @@ def sim_time_check(
     Returns:
         True if _any_ simulated task state has changed.
     """
-    sim_task_state_changed = False
-    now = time()
+    sim_task_state_changed: bool = False
     for itask in itasks:
-        if itask.state.status != TASK_STATUS_RUNNING:
+        if (
+            itask.state.status != TASK_STATUS_RUNNING
+            or itask.state.is_queued
+            or itask.state.is_held
+            or itask.state.is_runahead
+        ):
             continue
-        # Started time is not set on restart
-        if itask.summary['started_time'] is None:
-            itask.summary['started_time'] = now
-        timeout = (
-            itask.summary['started_time'] +
-            itask.tdef.rtconfig['simulation']['simulated run length']
-        )
-        if now > timeout:
+
+        # Started time and mode_settings are not set on restart:
+        started_time = itask.summary['started_time']
+        if started_time is None:
+            started_time = int(
+                TimePointParser()
+                .parse(
+                    db_mgr.pub_dao.select_task_job(
+                        *itask.tokens.relative_id.split("/")
+                    )["time_submit"]
+                )
+                .seconds_since_unix_epoch
+            )
+            itask.summary['started_time'] = started_time
+        if itask.mode_settings is None:
+            itask.mode_settings = ModeSettings(itask, broadcast_mgr)
```

Comment:
The `ModeSettings` gets set whenever it isn't set:

* This means it should persist for the life of a job which is correct.
  * E.G. a broadcast made whilst the simulated job is running won't effect the job
* This means it should persist for future submissions of the job which is incorrect.
  *  E.G. a broadcast made whilst a simulated job is running or after it has completed will not affect future submissions (i.e. auto-retries or manual triggers)

Could do with tests for reload/restart interactions to ensure behaviour is as expected.

Note: On restart, I think we re-submit the simulated job with the new config as we don't currently have the ability to "continue" it with the original config, this is fine.

Reply:
> This means it should persist for future submissions of the job which is incorrect

Bother. Will fix.

> Could do with tests for reload/restart interactions to ensure behaviour is as expected.

_**YES**_

> Note: On restart, I think we re-submit the simulated job with the new config as we don't currently have the ability to "continue" it with the original config, this is fine.

I'm not convinced, will investigate.

Reply:
> Note: On restart, I think we re-submit the simulated job with the new config as we don't currently have the ability to "continue" it with the original config, this is fine.

I was initially thinking about the fact that we use the database to get the start time back, but that's literally the only thing we retrieve, everything else is reloaded from config, so I think that you are correct, and that this needs to be a resubmission. 

Are we OK with it using bog-standard re-submission behavior where it increments the job number? This may cause confusion if users have set 'fail try 1 only' or were expecting the job to succeed first time. (I'll make sure I put in some explanatory Logging.)

Should we handle the abandoned submission?  Is there anything in the database/datastore which is likely to look confusingly incomplete from such a task?

Reply:
* When you restart a workflow, any simulated jobs which were running at the time the workflow stopped should continue to run (i.e. they shouldn't fail as the result of the restart).
* Ideally they would continue with the config they started with (i.e. the simulated job should run for the length of the configured runtime it was submitted with), but it's not a biggie if they continue with the config the workflow was restarted with (call it a documented limitation).
* The submit number should not change for a job which has "continued" in this way (i.e. it should behave just like a background task would if it was left running whilst the workflow restarted).

I think this is how it's currently implemented (this branch).

Reply:
It is, and If you're happy with the limitation you've described in this. Having said which, what if we put the timeout into the database as the end time? Then we can get the timeout on restart. That just leaves the intended target task state.

Reply:
I think that https://github.com/cylc/cylc-flow/pull/5721/commits/35b637782ab6754153700d025cab078e5bb7a10d answers this issue, both in the change and the testing. Will wait for the CI before I un-draft. Will spin the conversation into a new ticket.

Reply:
https://github.com/cylc/cylc-flow/issues/5942